packages/desktop-client/src/components/reports/reports/Calendar.tsx

```diff
@@ -0,0 +1,958 @@
+        if (earliestMonth > yearAgo) {
+          earliestMonth = yearAgo;
+        }
+
+        const allMonths = monthUtils
+          .rangeInclusive(earliestMonth, monthUtils.currentMonth())
+          .map(month => ({
+            name: month,
+            pretty: monthUtils.format(month, 'MMMM, yyyy'),
+          }))
+          .reverse();
+
+        setAllMonths(allMonths);
+      } catch (error) {
+        console.error('Error fetching earliest transaction:', error);
+      }
+    }
+    run();
+  }, []);
+
+  const dispatch = useDispatch();
+  const navigate = useNavigate();
+  const { isNarrowWidth } = useResponsive();
+  const title = widget?.meta?.name || t('Calendar');
+  const table = useRef(null);
+  const dateFormat = useDateFormat();
+
+  const onSaveWidgetName = async (newName: string) => {
+    if (!widget) {
+      throw new Error('No widget that could be saved.');
+    }
+
+    const name = newName || t('Calendar');
+    await send('dashboard-update-widget', {
+      id: widget.id,
+      meta: {
+        ...(widget.meta ?? {}),
+        name,
+      },
+    });
+  };
+
+  function onChangeDates(start: string, end: string, mode: TimeFrame['mode']) {
+    setStart(start);
+    setEnd(end);
+    setMode(mode);
+  }
+
+  async function onSaveWidget() {
+    if (!widget) {
+      throw new Error('No widget that could be saved.');
+    }
+
+    try {
+      await send('dashboard-update-widget', {
+        id: widget.id,
+        meta: {
+          ...(widget.meta ?? {}),
+          conditions,
+          conditionsOp,
+          timeFrame: {
+            start,
+            end,
+            mode,
+          },
+        },
+      });
+      dispatch(
+        addNotification({
+          type: 'message',
+          message: t('Dashboard widget successfully saved.'),
+        }),
+      );
+    } catch (error) {
+      dispatch(
+        addNotification({
+          type: 'error',
+          message: t('Failed to save dashboard widget.'),
+        }),
+      );
+      console.error('Error saving widget:', error);
+    }
+  }
+  const { totalIncome, totalExpense } = useMemo(() => {
+    if (!data || !data.calendarData) {
+      return { totalIncome: 0, totalExpense: 0 };
+    }
+    return {
+      totalIncome: data.calendarData.reduce(
+        (prev, cur) => prev + cur.totalIncome,
+        0,
+      ),
+      totalExpense: data.calendarData.reduce(
+        (prev, cur) => prev + cur.totalExpense,
+        0,
+      ),
+    };
+  }, [data]);
+
+  const onSort = (headerClicked: string, ascDesc: 'asc' | 'desc') => {
+    if (headerClicked === sortField) {
+      setAscDesc(ascDesc);
+    } else {
+      setSortField(headerClicked);
+      setAscDesc('desc');
+    }
+  };
+
+  const onOpenTransaction = useCallback(
+    (transaction: TransactionEntity) => {
+      navigate(`/transactions/${transaction.id}`);
+    },
+    [navigate],
+  );
+
+  const CHEVRON_HEIGHT = 26;
```

Comment:
We should declare these outside the component

Reply:
done